### DISCUSSION TITLE
FreeDNS IP mismatch & failed certificate

### DISCUSSION THREAD

**Discussion by melanielcash** [NightscoutFoundation/xDrip#2934] [Q&A]
I need the information on the FAQ's page re [FreeDNS IP mismatch & failed certificate](https://navid200.github.io/xDrip/docs/Nightscout/FreeDNS_Min_Login.html), but when I click on the link, it takes me to a page about the minimum log-in.  I think the page is linked incorrectly?  If not, I can't quite understand how it helps me to resolve my IP mismatch. 

![image](https://github.com/NightscoutFoundation/xDrip/assets/56098966/3b4b47bd-07ac-4e69-8f8b-bcc76f550658)

**Comment by Navid200** [comment]
Thanks for the feedback.  I will address both the page and this issue you are having.

Would you please restart the server as a starting point?  https://navid200.github.io/xDrip/docs/Nightscout/Restart.html
Does it bring up  your Nightscout?  I don't think so.  Please let me know.

If it does not, would you please log into your FreeDNS account in a browser?  This, you can do using a real computer.  
When you log in, what do you see?  Does it not tell you that your account has gone dormant?

**Comment by melanielcash** [comment]
Thanks Navid.  I restarted the server, and no, it doesn't bring up Nightscout.  
I logged into the FreeDNS account in a browser and no, it doesn't say the account is dormant.  I only built it last night, and haven't been able to get it working yet at all.  When it was doing the Nightscout build phase 2 it came up with a red error saying Ping: unknown host, resolve call failed.  I think I must have done something wrong choosing a subdomain?  
<img width="573" alt="image" src="https://github.com/NightscoutFoundation/xDrip/assets/56098966/289e21db-42b6-4046-a765-2b601b3be50f">

**Reply by Navid200** [reply]
I see.
I didn't know that this was never working.
I will update the guide accordingly.

You are showing your hostname in your screenshot just in case you didn't mean to, you should delete it.

I would love to know exactly what happened.  This is obviously a fail that our code does not catch.
If I can recreate this problem, I will be able to add the intelligence to our code to detect and guide accordingly.

**Reply by melanielcash** [reply]
Thank you so much.  Is there anything I can do to try and help? I believe I followed the guide exactly, but I guess I can't have. Can I show some logs or something, or perhaps try and remember more details? 

I've added below the end of the phase 2 errors messages, in case that helps? 
<img width="656" alt="image" src="https://github.com/NightscoutFoundation/xDrip/assets/56098966/cc98bef3-02ab-498e-bb89-5751975b6fc7">

**Reply by Navid200** [reply]
When you log into your FreeDNS account and go to the subdomains page, do you see this subdomain in the list?

The next time you try, when you get to "What would you like to do", choose 2.
Is that what you chose?

**Reply by melanielcash** [reply]
Yes, I see it in the subdomains page.  
No, the first time I chose 1, I'll try 2 now.

**Reply by melanielcash** [reply]
I tried 2, it seemed to go ok, but when i clicked on the status page, get the same error.  
![image](https://github.com/NightscoutFoundation/xDrip/assets/56098966/f5508127-0f3a-4629-a7ae-06f0c30e1cb5)

**Reply by Navid200** [reply]
Do you mean phase 2 completed with no errors and it automatically restarted the server?
And this is what you see after the restart?

**Reply by melanielcash** [reply]
Apologies.  I chose option 2 as you'd mentioned above (to renew and replace the certificate), rather than 1 (to attempt to reinstall the certificate). That seemed to go ok with no errors, but when I went to the status page, the error was still there.  I'm trying phase 2 now, with exactly the same issue - ping: unknown host and the subdomain not found (all written in red).  

Should I perhaps try and create a new subdomain?  Could it be that I'm doing it on a work computer and even though its not blocked 'generally', perhaps my work computer isn't letting me go there?

**Reply by Navid200** [reply]
But, the virtual machine is in Google Cloud.  It is not at your place of work.  So, I don't see how that could be the cause of the problem.

Yes, please create a subdomain on a different public domain on FreeDNS and try that to see if the exact same problem happens again or not?
You can have a total of 5 different subdomains on as many as 5 different domains for free.

**Comment by melanielcash** [comment]
Ok, I tried on my home mac anyway, just in case.

I created a new subdomain on a different public domain, and a new virtual
machine, and got through with no errors. Awesome.

Just now having an issue with opening the webpage. When I go to open my
subdomain (I’ve checked it’s spelled exactly the same in the status page
and on the Google cloud console), it says 502 bad gateway. Any suggestions?

Mel

**Reply by Navid200** [reply]
What does the status page show?

**Comment by melanielcash** [comment]
Apologies, I followed your troubleshooting guide for the 502 error and it’s
working. Hooray!!

Thank you so much for your help, I would have been completely lost
otherwise.

**Reply by Navid200** [reply]
Great!
You're welcome.
